src/python/qequlacs/utils.py

```diff
@@ -38,31 +76,54 @@ def append_gate(gate, qulacs_circuit):
     elif len(gate.qubits) == 2:
         index_1 = gate.qubits[0].index
         index_2 = gate.qubits[1].index
+
         if gate.name == 'CNOT':
             qulacs_circuit.add_CNOT_gate(index_1, index_2)
         elif gate.name == 'SWAP':
             qulacs_circuit.add_SWAP_gate(index_1, index_2)
         elif gate.name == 'XX':
-            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [1,1], -gate.params[0]*2/np.pi)
+            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [1,1], -angle)
         elif gate.name == 'YY':
-            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [2,2], -gate.params[0]*2/np.pi)
+            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [2,2], -angle)
         elif gate.name == 'ZZ':
-            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [3,3], -gate.params[0]*2/np.pi)
+            qulacs_circuit.add_multi_Pauli_rotation_gate([index_1, index_2], [3,3], -angle)
         elif gate.name == 'CPHASE':
-            mat = np.diag([1., np.exp(1.j*np.pi*gate.params[0])])
-            gate = qulacs.gate.DenseMatrix(index_2, mat)
-            gate.add_control_qubit(index_1,1)
+            matrix = np.diag([1., np.exp(1.j * angle)])
+            gate = qulacs.gate.DenseMatrix(index_2, matrix)
+            gate.add_control_qubit(index_1, 1)
             qulacs_circuit.add_gate(gate)
+        elif gate.name == 'CZ':
+            qulacs_circuit.add_CZ_gate(index_1, index_2)
         else:
             unitary_matrix = gate.to_unitary()
             qulacs_circuit.add_dense_matrix_gate([index_1, index_2], unitary_matrix)
+
+    # Three qubit gate
+    elif len(gate.qubits) == 3:
+        index_1 = gate.qubits[0].index
+        index_2 = gate.qubits[1].index
+        index_3 = gate.qubits[2].index
+
+        if gate.name == 'CSWAP':
+            # Fredkin gate
+            gate = qulacs.gate.FREDKIN(index_1, index_2, index_3)
+            qulacs_circuit.add_gate(gate)
+        elif gate.name == 'CCNOT':
+            # Toffoli gate
+            gate = qulacs.gate.TOFFOLI(index_1, index_2, index_3)
+            qulacs_circuit.add_gate(gate)
+        else:
+            unitary_matrix = gate.to_unitary()
+            qulacs_circuit.add_dense_matrix_gate([index_1, index_2, index_3], unitary_matrix)
     else:
         n_qubits = len(gate.qubits)
         Exception(str(n_qubits) + "- qubit gates not supported.")
     return qulacs_circuit
 
 
 def qubitop_to_qulacspauli(qubit_operator):
```

Comment:
I've also noticed we lack typing in this module.
`qubit_operators` should be of type `openfermion.QubitOperator`